### PR TITLE
Cpu support of batchnorm layernorm module

### PR DESCRIPTION
**Pull request by Flowingsun007**
- layernorm、batchnorm module增加对cpu的支持；
- 新增了flow.var module

- docs:
flow var

batchnorm

### CODE REVIEW

File: oneflow/python/nn/modules/batchnorm.py [hunk 1]
```diff
@@ -20,6 +20,78 @@
...
+        else:
+            if weight is None:
+                weight = flow.experimental.ones(size=params_shape, dtype=params_dtype)
+            if bias is None:
+                bias = flow.experimental.zeros(size=params_shape, dtype=params_dtype)
+            res = self._op(x, mean, variance, weight, bias)[0]
+            return res
+
+
+@oneflow_export("batch_normalization")
```
Comment (Ldpe2G) [inline oneflow/python/nn/modules/batchnorm.py:87]: 为什么要再加一个 batch_normalization 得实现，而且 Pytorch 没有这个接口的

File: oneflow/python/nn/modules/batchnorm.py [hunk 2]
```diff
@@ -38,18 +110,18 @@ def __init__(
         self.affine = affine
         self.track_running_stats = track_running_stats
         if self.affine:
-            self.weight = flow.nn.Parameter(flow.Tensor(num_features))
-            self.bias = flow.nn.Parameter(flow.Tensor(num_features))
+            self.weight = flow.nn.Parameter(
```
Comment (Ldpe2G) [inline oneflow/python/nn/modules/batchnorm.py:113]: 下面参数和buffer的初始化改动是多余的，下面 self.reset_parameters 就会做初始化

File: oneflow/python/nn/modules/reduce_ops.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+    def forward(self, input):
+        axis = _check_axis(self.axis, input.shape)
+        if len(axis) == 0:
+            return input
+        else:
+            return self._op(input, axis=axis)[0]
+
+
+@oneflow_export("reduce_sum")
```
Comment (Ldpe2G) [inline oneflow/python/nn/modules/reduce_ops.py:46]: reduce_sum 在 math_ops.py 已经提供了，而且 Pytorch 也没有 reduce_sum 的 module

File: oneflow/python/nn/modules/reduce_ops.py [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        np_arr = np.random.randn(2,3,4,5)
+        input = flow.Tensor(np_arr)
+        output = flow.reduce_sum(input, 1, True)
+        # equal to np.sum(np_arr, 1, keepdims=True)
+
+    """
+    return ReduceSum(axis, keepdims)(input)
+
+
+@oneflow_export("reduce_mean")
```
Comment (Ldpe2G) [inline oneflow/python/nn/modules/reduce_ops.py:75]: reduce_mean 在 math_ops.py 已经提供了，而且 Pytorch 也没有 reduce_mean 的 module

File: oneflow/python/nn/modules/reduce_ops.py [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+                flow.experimental.reduce_mean(
+                    flow.experimental.square(input), axis, self.keepdims
+                ),
+                flow.experimental.square(
+                    flow.experimental.reduce_mean(input, axis, self.keepdims)
+                ),
+            )
+
+
+@oneflow_export("reduce_variance")
```
Comment (Ldpe2G) [inline oneflow/python/nn/modules/reduce_ops.py:136]: 这个var放到 math_ops.py 里面吧，而且注册的名字应该是 var 和 pytorch 一致

**Comment by Ldpe2G** [comment]
而且单测好像只测试了cpu，因为现在默认是跑的cpu，是不是得加上to到gpu的测试

File: oneflow/python/nn/modules/batchnorm.py (hunk 1 above)
Comment (Flowingsun007) [reply oneflow/python/nn/modules/batchnorm.py]: 我把这个batch_normalization取消导出了（用这个是因为batchnorm、layernorm的cpu实现都依赖这个

File: oneflow/python/nn/modules/batchnorm.py (hunk 2 above)
Comment (Flowingsun007) [reply oneflow/python/nn/modules/batchnorm.py]: 已恢复

File: oneflow/python/nn/modules/reduce_ops.py (hunk 3 above)
Comment (Flowingsun007) [reply oneflow/python/nn/modules/reduce_ops.py]: 已修改

File: oneflow/python/nn/modules/reduce_ops.py (hunk 4 above)
Comment (Flowingsun007) [reply oneflow/python/nn/modules/reduce_ops.py]: 已修改

File: oneflow/python/nn/modules/reduce_ops.py (hunk 5 above)
Comment (Flowingsun007) [reply oneflow/python/nn/modules/reduce_ops.py]: 已移至math_ops.py

**Review by Ldpe2G — APPROVED** [review]
